proto/spire/next/api/entry/v1/entry.proto

```diff
@@ -37,47 +38,85 @@ message ListEntriesRequest {
         string by_spiffe_id = 3;
     }
 
-    // Filters the entries returned by the list operation.
+    // Filters the entries returned in the response.
+    //
+    // TODO: How to document optional fields?
     Filter filter = 1;
 
-    // The maximum number of items to return.
-    int32 page_size = 2;
+    // Masks the entry fields returned in the response.
+    spire.types.EntryMask mask = 2;
+
+    // The maximum number of entries to return.
+    int32 page_size = 3;
 
-    // The next_page_token value returned from a previous List request, if any.
-    string page_token = 3;
+    // The `next_page_token` value obtained from the previous `ListEntriesResponse` message, if any.
+    string page_token = 4;
 }
 
 message ListEntriesResponse {
+    // The list of entries.
     repeated spire.types.Entry entries = 1;
 
+    // The token to use as the value for `page_token` in the next `ListEntriesRequest`.
     string next_page_token = 2;
 }
 
 message GetEntryRequest {
+    // ID of the entry to get.
     string id = 1;
+
+    // Masks the entry fields returned in the response.
+    spire.types.EntryMask mask = 2;
 }
 
+// TODO: Do we never need a response message? CreateEntryIfNotExists needed
+// one to tell the caller whether the record was preexisting or not.
+//
 message CreateEntryRequest {
+    // The entry to be created. If no entry ID is set, one will be generated.
     spire.types.Entry entry = 1;
+
+    // Masks the entry fields returned in the response.
+    spire.types.EntryMask mask = 2;
 }
 
 message BatchCreateEntryRequest {
+    // The entries to be created. IDs will be generated for entries that don't have one set.
     repeated spire.types.Entry entries = 1;
+
+    // Masks the entry fields returned in the response.
+    spire.types.EntryMask mask = 2;
 }
 
 message BatchCreateEntryResponse {
+    // The entries that were created.
     repeated spire.types.Entry entries = 1;
+
+    // TODO: error map!
 }
 
 message UpdateEntryRequest {
+    // The updated entry.
     spire.types.Entry entry = 1;
-    spire.types.EntryMask mask = 2;
+
+    // Masks the entry fields returned in the response
+    spire.types.EntryMask mask = 3;
 }
 
 message DeleteEntryRequest {
+    // ID of the entry to delete.
     string id = 1;
 }
 
 message BatchDeleteEntryRequest {
+    // IDs of the entries to delete.
     repeated string id = 1;
 }
+
+message BatchDeleteEntryResponse {
+    // Errors that occured while deleting the requested entries, if any, keyed
+    // by entry ID.
+    //
+    // TODO: will this broadly work?
```

Comment:
A pattern that would work relatively consistently would be great. Same problem in Batch Create, but with a slightly different shape.

I couldn't find any established patterns for this, but maybe I didn't look in the right places

Reply:
Here's an example:

https://github.com/googleapis/googleapis/blob/59b559242c6fd9dc43e958043141a64c9e70dba9/google/streetview/publish/v1/rpcmessages.proto#L249

Instead of a sparsely populated map, they provide a repeated field for status, one for each id to delete.

Reply:
For BatchCreate and BatchUpdate, I'd assume we'd want some repeated message that holds onto a status and an entry (my chosen name is terrible):
```
message EntryAndStatus {
    google.rpc.Status = 1;
    spire.type.Entry entry = 2;
}

message BatchCreateEntryResponse {
    repeated EntryStatus status = 1;
}
```

Reply:
From the example, it looks like the contract is that a response is always included, with the index number corresponding to that of the original request it represents.

That seems to work pretty well in the general case of responding to a repeated batch request, but as you noted it doesn't quite fit Create. Interestingly, the proto you linked is also missing a Batch RPC for Create...